content/blog/2017/2017-02-02-scm-api-2-take2.adoc

```diff
@@ -0,0 +1,116 @@
+Consequently, in Branch API 2.0.0 the names were all mangled into URL safe names, with the original URLs still resolving to the original projects so that any existing saved links would remain functional.
++
+Quite a number of people objected to the change of URL scheme.
++
+*Resolution:* There has been a convention in Jenkins that the on-disk storage structure for jobs mirrors the URL structure.
+This is only a convention and there is nothing specific in the code that mandates following the convention.
++
+The Folders Plugin was enhanced to allow for computed folders (where the item names are provided by an external source) to provide a strategy to use when generating the on-disk storage names as well as the URL component names for the folder's child items.
++
+The Branch API plugin was enhanced to use this new strategy for name mangling.
++
+The net effect of this change is that the URLs remain the same as for 1.x but the on-disk storage uses mangled names that are future proofed against any new SCMNavigator implementations where the backing service allows names that are problematic to use as filesystem directory names.
++
+*Side-effect:* The Branch API 2.0.0 approach handled the mangling of names by renaming the items using the Jenkins Item rename API.
+The Branch API 2.0.2 approach does not rename the child items as it is only the on-disk storage location that is moved.
+This means that the Jenkins Item rename API cannot be used.
++
+At this time, the only known side-effect is in the  link:https://plugins.jenkins.io/jobConfigHistory[Job Configuration History] plugin.
+The configuration history of each child item will still be tracked going forward after the upgrade.
+The pre-upgrade configuration history is also retained.
+Because the Jenkins Item rename API cannot be used to flag the configuration file location change, there is no association between the pre-upgrade history chain and the post-upgrade history chain.
+
+These issues are all resolved with the following plugin releases:
+
+Folders Plugin:: *5.17*
+SCM API Plugin:: *2.0.2*
+Branch API Plugin:: *2.0.2*
+Git Plugin:: *Either* *2.6.4* *or* *3.0.4*
+GitHub Branch Source Plugin:: *2.0.1*
+BitBucket Branch Source Plugin:: *2.0.2*
+Pipeline Multibranch Plugin:: *2.12*
+
+So here is the full details on plugin versions:
+
+== Summary for busy Jenkins Administrators
+
+Upgrading should make multi-branch projects much better.
+When you are ready to upgrade you must ensure that you upgrade all the required plugins.
+If you miss some, just upgrade them and restart to fix the issue.
+
+Folders Plugin:: *5.17* or newer
+SCM API Plugin:: *2.0.2* or newer
+Branch API Plugin:: *2.0.2* or newer
+Git Plugin:: *Either* *2.6.4* or newer in the 2.6.x line *or* *3.0.4* or newer
+Mercurial Plugin:: 2.0.0 or newer
+GitHub Branch Source Plugin:: *2.0.1* or newer
+BitBucket Branch Source Plugin:: *2.0.2* or newer
+GitHub Organization Folders Plugin:: 1.6
+Pipeline Multibranch Plugin:: *2.12* or newer
+
```

Comment:
This list is the full list of all plugins from the original blog post with the versions updated.

The other list (now at the bottom of the post) is the list of plugins that we updated